src/duckling/game/collision/collision.component.ts

```diff
@@ -68,7 +100,17 @@ export class CollisionComponent {
         this.attributeChanged.emit(immutableAssign(this.attribute, {bodyType}));
     }
 
+    onEditCollisionTypesClicked() {
+        EditCollisionTypesComponent.open(this._viewContainer);
+    }
+
     onCollisionTypeInput(collisionType : string) {
         this.attributeChanged.emit(immutableAssign(this.attribute, {collisionType}));
     }
+
+    private _buildCollisionTypeOptions() : SelectOption[] {
+        let selectOptions : SelectOption[] = [];
+        this._collisionTypes.collisionTypes.getValue().map(collisionType => selectOptions.push({value: collisionType, title: collisionType}));
```

Comment:
This is not idiomatic usage of map. You should return the value instead of mutating selectOptions.